sdk/core/Azure.Core/tests/ClientDiagnosticsTests.Net50.cs

```diff
@@ -99,6 +99,168 @@ public void StartsActivitySourceActivity()
             }
         }
 
+        [TestCase(null)]
+        [TestCase(true)]
+        [NonParallelizable]
+        public void NestedClientActivitiesConfigurationClientOptions(bool? enableNestedScopes)
+        {
+            // TODO: ClientDiagnostics need new Azure.Core to ship before using
+            // DiagnosticsOptions.AreNestedClientSpansEnabled flag so we can't check it yet.
+
+            using var testListener = new TestDiagnosticListener("Azure.Clients");
+            DiagnosticsOptions testOptions = new DiagnosticsOptions();
+            if (enableNestedScopes.HasValue)
+            {
+                testOptions.IsNestingClientSpansEnabled = enableNestedScopes.Value;
+            }
+
+            ClientDiagnostics clientDiagnostics = new ClientDiagnostics("Azure.Clients", "Microsoft.Azure.Core.Cool.Tests", testOptions);
+
+            using DiagnosticScope scope = clientDiagnostics.CreateScope("ClientName.ActivityName");
+            scope.Start();
+
+            DiagnosticScope nestedScope = clientDiagnostics.CreateScope("ClientName.NestedActivityName");
+            nestedScope.Start();
+            if (enableNestedScopes.GetValueOrDefault(true))
+            {
+                Assert.IsTrue(nestedScope.IsEnabled);
+                Assert.AreEqual("ClientName.NestedActivityName", Activity.Current.OperationName);
+            }
+            else
+            {
+                Assert.IsFalse(nestedScope.IsEnabled);
+                Assert.AreEqual("ClientName.ActivityName", Activity.Current.OperationName);
+            }
+
+            nestedScope.Dispose();
+            Assert.AreEqual("ClientName.ActivityName", Activity.Current.OperationName);
+        }
+
+        [TestCase(0, true)]  // Internal
```

Comment:
You can use enum directly:

```cs
[TestCase(DiagnosticScope.ActivityKind.Internal, true)]
[TestCase(DiagnosticScope.ActivityKind.Server, false)]
[TestCase(DiagnosticScope.ActivityKind.Client, true)]
[TestCase(DiagnosticScope.ActivityKind.Producer, false)]
[TestCase(DiagnosticScope.ActivityKind.Consumer, false)]
```

Reply:
`DiagnosticScope.ActivityKind` is internal, and the test method is public, so no, I can't

Reply:
![image](https://user-images.githubusercontent.com/8933858/163865555-4925f282-87d4-4e47-9d56-45faeba35e89.png)